solidity/test/bridge/Bridge.MovingFunds.test.ts

```diff
@@ -1455,6 +1455,149 @@ describe("Bridge - Moving funds", () => {
+    const walletDraft = {
+      ecdsaWalletID: ecdsaWalletTestData.walletID,
+      mainUtxoHash: ethers.constants.HashZero,
+      pendingRedemptionsValue: 0,
+      createdAt: 0,
+      movingFundsRequestedAt: 0,
+      state: walletState.Unknown,
+      movingFundsTargetWalletsCommitmentHash: ethers.constants.HashZero,
+    }
+
+    context("when source wallet is in the MovingFunds state", () => {
+      before(async () => {
+        await createSnapshot()
+
+        await bridge.setWallet(ecdsaWalletTestData.pubKeyHash160, {
+          ...walletDraft,
+          movingFundsRequestedAt: await lastBlockTime(),
+          state: walletState.MovingFunds,
+        })
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      context("when the moving funds process has timed out", () => {
+        let tx: ContractTransaction
+
+        before(async () => {
+          await createSnapshot()
+
+          await increaseTime(
+            (
+              await bridge.movingFundsParameters()
+            ).movingFundsTimeout
+          )
+
+          tx = await bridge.notifyMovingFundsTimeout(
+            ecdsaWalletTestData.pubKeyHash160
+          )
+        })
+
+        after(async () => {
+          walletRegistry.closeWallet.reset()
+
+          await restoreSnapshot()
+        })
+
+        it("should switch the wallet to Terminated state", async () => {
+          expect(
+            (await bridge.wallets(ecdsaWalletTestData.pubKeyHash160)).state
+          ).to.be.equal(walletState.Terminated)
+        })
+
+        it("should emit WalletTerminated event", async () => {
+          await expect(tx)
+            .to.emit(bridge, "WalletTerminated")
+            .withArgs(
+              ecdsaWalletTestData.walletID,
+              ecdsaWalletTestData.pubKeyHash160
+            )
+        })
+
+        it("should call ECDSA Wallet Registry's closeWallet function", async () => {
+          // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+          expect(walletRegistry.closeWallet).to.have.been.calledOnceWith(
+            ecdsaWalletTestData.walletID
+          )
+        })
+
+        it("should emit MovingFundsTimedOut event", async () => {
+          await expect(tx)
+            .to.emit(bridge, "MovingFundsTimedOut")
+            .withArgs(ecdsaWalletTestData.pubKeyHash160)
+        })
+
+        it("should slash wallet operators", async () => {
+          // TODO: Implementation once slashing is integrated.
+        })
+
+        it("should reward the notifier", async () => {
+          // TODO: Implementation once slashing is integrated.
+        })
+      })
+
+      context("when the moving funds process has not timed out", () => {
+        it("should revert", async () => {
+          await expect(
+            bridge.notifyMovingFundsTimeout(ecdsaWalletTestData.pubKeyHash160)
+          ).to.be.revertedWith("Moving funds has not timed out")
+        })
+      })
```

Comment:
In most other test cases where we test timeout notifications, we increase time by the `timeout` value minus one second.
We could consider using this pattern here.

Reply:
Yeah, see https://github.com/keep-network/tbtc-v2/pull/212/commits/72832423a72d567caf23f79dc53216c6f6245a8b